map/script.js

```diff
@@ -0,0 +1,16 @@
+const array = [1, 1, 2, 3, 1, 3, 1];
+
+console.log(array);
+
+let arrayMap = new Map();
+let currentValue;
+
+for(let i = 0; i < array.length; i++) {
+    currentValue = arrayMap.get(array[i])

```

Comment:
As we calculate currentValue inside 'for' - better to declare it inside 'for' rather than outside.